# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/generic/LocalAccessible.h
# Commit: 6591f14bbc71
# Full Hash: 6591f14bbc713f11b6316712d931ecc19a443a5c
# Author: Morgan Reschenberg <mreschenberg@mozilla.com>
# Date: 2021-10-05 03:52:32
# Regressor Bug: 1726227
# File Overlap Count: 1
# Description:
#   Bug 1726227: Cache parent-relative accessible bounds, resolution in parent process r=Jamie,eeejay
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D123399
# ==============================================================================

diff -r 06dda9072cb7 -r 6591f14bbc71 accessible/generic/LocalAccessible.h
--- a/accessible/generic/LocalAccessible.h	Mon Oct 04 19:28:49 2021 +0000
+++ b/accessible/generic/LocalAccessible.h	Mon Oct 04 19:46:54 2021 +0000
@@ -478,7 +478,7 @@
   /**
    * Return boundaries in screen coordinates.
    */
-  virtual nsIntRect Bounds() const;
+  virtual nsIntRect Bounds() const override;
 
   /**
    * Return boundaries in screen coordinates in CSS pixels.
@@ -491,6 +491,11 @@
   virtual nsRect RelativeBounds(nsIFrame** aRelativeFrame) const;
 
   /**
+   * Return boundaries rect relative to the frame of the parent accessible.
+   */
+  virtual nsRect ParentRelativeBounds();
+
+  /**
    * Selects the accessible within its container if applicable.
    */
   virtual void SetSelected(bool aSelect);
@@ -1079,6 +1084,7 @@
   LocalAccessible* mParent;
   nsTArray<LocalAccessible*> mChildren;
   int32_t mIndexInParent;
+  Maybe<nsRect> mBounds;
 
   static const uint8_t kStateFlagsBits = 11;
   static const uint8_t kContextFlagsBits = 3;